docs/setup.rst

```diff
@@ -0,0 +1,79 @@
+Setting Up Corobo
+=================
+
+The following documentation illustrates the process to install corobo and interface it with the gitter(or any other backend) API.
+
+Installation
+------------
+
+*  To install corobo, first you need to setup a virtual environment(it is not mandatory but highly recommended). This can be done as follows:
+   ::
+     pip3 install virtualenv
+     virtualenv venv -p python
+     cd venv
+     source bin/activate ---->  For Linux
+     \Scripts\activate   ---->  For Windows
+
+*  Next, you can clone the corobo package via git and install the requirements(dependent packages).
+   ::
+     git clone https://github.com/coala/corobo.git
+     pip install -r requirements.txt
+     
+     
+Configuring the backend
+------------------------
+
+#. Clone the err-backend-gitter repository in your "corobo" folder.
+   ::
+     git clone https://github.com/errbotio/errbot@a0f35732484c8c0692e123c48653517cffa21a42
+          
+
+#. To make the bot interact with the gitter API you will need a github account(this account can be used to login to gitter) to generate the gitter access token.
+   **Remember it is recommended to use a separate account(other than the admin account) for the bot.** 
+   Once the account is setup, run the following script.
+   ::
+     ./oauth.py
+   This will guide you with the gitter token generation process.
+
+#. Keep this token safe. It will be used later to set the environment variables.
+     
+#. Also to configure the bot with access to your github and gitlab repositories, you will need personal access tokens.
+   The following links may guide you through this:
+     https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token
+     https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html
+
+#. You can also create github organisations and gitlab groups to manage your repositories.
+   Under a organisation, you can make teams to organize the workforce.
```

Comment:
Either use American English or British, not both. I'd suggest running this text though a spell and grammar checker.